hearth/spec/hearth/http/middleware/request_compression_spec.rb

```diff
@@ -81,24 +81,59 @@ def expect_uncompressed_body(request, body)
             end
           end
 
-          context 'no supported encodings' do
-            let(:encodings) { ['custom'] }
+          context 'encodings' do
+            context 'none supported by client' do
+              let(:encodings) { ['custom'] }
 
-            it 'skips compression' do
-              expect(app).to receive(:call).with(input, context)
-              resp = subject.call(input, context)
-              expect_uncompressed_body(request, body)
-              expect(resp).to be output
+              it 'skips compression' do
+                expect(app).to receive(:call).with(input, context)
+                resp = subject.call(input, context)
+                expect_uncompressed_body(request, body)
+                expect(resp).to be output
+              end
             end
-          end
 
-          context 'multiple encodings' do
-            let(:encodings) { %w[custom gzip] }
+            context 'multiple encodings' do
+              let(:encodings) { %w[custom gzip] }
+              let(:body) { StringIO.new('a' * 16_385) }
+
+              it 'processes the first supported encoding found' do
+                expect(app).to receive(:call).with(input, context)
+                resp = subject.call(input, context)
+                expect(request.headers['Content-Encoding']).to eq('gzip')
+                expect(resp).to be output
+              end
+            end
+
+            context 'unsupported encoding is given' do
+              it 'raises a standard error in compression' do
+                expect do
+                  subject.send(
+                    :process_compression,
+                    'custom',
+                    request
+                  )
+                end.to raise_error(StandardError)
```

Comment:
I think these errors should have been RuntimeErrors - I see it's StandardError in the implementation though.

Reply:
I just did some digging on this to understand better. Makes sense to use `RuntimeError` as it is a reasonable default rather than using the parent class, `StandardError`. 

I'll change it now.

Reply:
Nit pick on the error message too.. Not sure if we should use language like "We do not" - try to be objective - something like "Encoding #{encoding} is not supported"

Reply:
This error should never actually reach end users - the modeling validation ensures that only a set of supported encodings is applied and the logic above in selecting a supported algorithm ensures we would never get here... hence the difficulty in testing it directly.

We could just remove this case completely.

Reply:
Per our discussion, I will be removing this case and any specs related to it completely